code/modules/unit_tests/objectives.dm

```diff
@@ -9,15 +9,26 @@
 			else
 				objectives_that_exist += value
 
+	SStraitor.generate_objectives = TRUE
```

Comment:
What is this? This looks really weird. Also make sure to untick it on Destroy instead.

Reply:
It's explained in the comments of the variable. Also I accidentally got the order mixed, that should be set to FALSE and then set to TRUE
/// Used by the debug menu, decides whether newly created objectives should generate progression and telecrystals. Do not modify for non-debug purposes.
It could be better named, but it basically just determines whether the traitor objective should initialize its random values on creation or not. It allows for you to obtain an instance of a traitor objective where the tc reward and progression reward are in their list form if they're supposed to vary. It's useful to the debug menu, but it'll possibly stop random CI failures if you have an objective with a telecrystal reward and progression reward like this:
```dm
progression_reward = list(0, 2 MINUTES)
telecrystal_reward = list(0, 1)
```

Reply:
I see. You're testing the list values and you can't use initial because of that